Source/DataSources/KmlDataSource.js

```diff
@@ -818,10 +818,17 @@ define([
         //Google earth seems to always use the first external style only.
         var externalStyle = queryStringValue(placeMark, 'styleUrl', namespaces.kml);
         if (defined(externalStyle)) {
-            //Google Earth ignores leading and trailing whitespace for styleUrls
-            //Without the below trim, some docs that load in Google Earth won't load
-            //in cesium.
-            var id = externalStyle;
+            var id;
+            if (externalStyle[0] !== '#' && externalStyle.indexOf('#') !== -1) {
+                var tokens = externalStyle.split('#');
+                var uri = tokens[0];
+                if (defined(sourceUri)) {
+                    uri = getAbsoluteUri(uri, getAbsoluteUri(sourceUri));
+                }
+                id = uri + '#' + tokens[1];
+            } else {
+                id = externalStyle;
```

Comment:
nitpick,  but I would just do `var id = externalStyle;` above and remove this else block.